script/build.sh

```diff
@@ -3,44 +3,100 @@
 set -e
 
 DIR="$(cd "$(dirname "$0")/../" && pwd)"
+INCLUDE_DIR=$DIR
+SOURCE_DIR=$DIR
+BUILD_DIR=$DIR/build
+BUILD_INT_DIR=$BUILD_DIR/intermediate
+BUILD_BIN_DIR=$BUILD_DIR/bin
+STATIC_BUILD_INT_DIR=$BUILD_DIR/intermediate/static
+SHARED_BUILD_INT_DIR=$BUILD_DIR/intermediate/shared
+BUILD_LIB_DIR=$BUILD_DIR/lib
+STATIC_BUILD_LIB_DIR=$BUILD_LIB_DIR/static
+SHARED_BUILD_LIB_DIR=$BUILD_LIB_DIR/shared
+TEST_SOURCE_DIR=$DIR/test
+TEST_INT_DIR=$BUILD_INT_DIR/test
+TEST_BIN_DIR=$BUILD_BIN_DIR/test
+EXAMPLES_SOURCE_DIR=$DIR/examples
+EXAMPLES_INT_DIR=$BUILD_INT_DIR/examples
+EXAMPLES_BIN_DIR=$BUILD_BIN_DIR/examples
+
+CFLAGS="-I$INCLUDE_DIR"
+CXXFLAGS="-I$INCLUDE_DIR"
+LDFLAGS=""
 
 if [ "$(uname)" = "Darwin" ]; then
-	FLAGS="-DWEBVIEW_COCOA -std=c++11 -Wall -Wextra -pedantic -framework WebKit"
+	LDFLAGS="-framework WebKit"
+	CXXFLAGS="$CXXFLAGS -DWEBVIEW_COCOA -std=c++11 -Wall -Wextra -pedantic"
 else
-	FLAGS="-DWEBVIEW_GTK -std=c++11 -Wall -Wextra -pedantic $(pkg-config --cflags --libs gtk+-3.0 webkit2gtk-4.0)"
+	LDFLAGS="$(pkg-config --libs gtk+-3.0 webkit2gtk-4.0)"
+	CXXFLAGS="$CXXFLAGS -DWEBVIEW_GTK -std=c++11 -Wall -Wextra -pedantic $(pkg-config --cflags gtk+-3.0 webkit2gtk-4.0)"
 fi
 
 if command -v clang-format >/dev/null 2>&1 ; then
 	echo "Formatting..."
-	clang-format -i \
-		"$DIR/webview.h" \
-		"$DIR/webview_test.cc" \
-		"$DIR/main.cc"
+	# TODO: Clean up examples and remove them from the exclusion list
```

Comment:
I will do this work if and when this PR gets merged as to not add noise to this PR, since the examples were not checked before and have some issues.